src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AbstractSubmitToArchiveCommand.java

```diff
@@ -0,0 +1,99 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.DatasetVersion;
+import edu.harvard.iq.dataverse.DvObject;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.authorization.users.ApiToken;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.engine.command.Command;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStepResult;
+
+import static edu.harvard.iq.dataverse.engine.command.CommandHelper.CH;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+@RequiredPermissions(Permission.PublishDataset)
+public abstract class AbstractSubmitToArchiveCommand implements Command<DatasetVersion> {
```

Comment:
@qqmyers  Any reason not to extend AbstractCommand (and then you wouldn't have to override so many methods?)

Reply:
DatasetVersion is not a DvObject so AbstractCommand doesn't quite fit.

Reply:
Hmm, good point. I wonder if you could do something like this then (pass version number):
https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/GetSpecificPublishedDatasetVersionCommand.java

It could even have several constructors (one with version numbers, one with version ids).

And let the command get the version.

Alternatively, it could make sense to modify AbstractCommand to be more flexible (not necessarily in scope for this issue)

Reply:
Possibly. That wouldn't log the version number without modifying AbstractCommand though. And since the api call and future GUI start from the version, they'd have to find the dataset to send just so the version could be looked up again. Just brainstorming - some other redesign that let you put any/all params into a Map (whether DvObject or not) with AbstractCommand printing simple params like longs and calling callbacks on other objects to have them print themselves might be useful...

Reply:
OK, so I found an example of a command that does take a DataseVersion:
https://github.com/IQSS/dataverse/blob/develop/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/DeaccessionDatasetVersionCommand.java

Could you mimic that?